test/test_grid.py

```diff
@@ -761,27 +850,55 @@ def test_build_face_edges_connectivity_fillvalues(self):
             self.f0_deg, self.f1_deg, self.f2_deg, self.f3_deg, self.f4_deg,
             self.f5_deg, self.f6_deg
         ]
-        uds = ux.open_grid(verts)
+        uds = ux.Grid(verts)
         uds._build_face_edges_connectivity()
-        n_face = len(uds._ds["Mesh2_face_edges"].values)
+        n_face = len(uds.ds["Mesh2_face_edges"].values)
         n_node = uds.nMesh2_node
-        n_edge = len(uds._ds["Mesh2_edge_nodes"].values)
+        n_edge = len(uds.ds["Mesh2_edge_nodes"].values)
 
         self.assertEqual(7, n_face)
         self.assertEqual(21, n_node)
         self.assertEqual(28, n_edge)
 
         # We will utilize the edge_nodes_connectivity and face_edges_connectivity to generate the
-        # res_face_nodes_connectivity and compare it with the uds._ds["Mesh2_face_nodes"].values
-        edge_nodes_connectivity = uds._ds["Mesh2_edge_nodes"].values
-        face_edges_connectivity = uds._ds["Mesh2_face_edges"].values
-        face_nodes_connectivity = uds._ds["Mesh2_face_nodes"].values
+        # res_face_nodes_connectivity and compare it with the uds.ds["Mesh2_face_nodes"].values
+        edge_nodes_connectivity = uds.ds["Mesh2_edge_nodes"].values
+        face_edges_connectivity = uds.ds["Mesh2_face_edges"].values
+        face_nodes_connectivity = uds.ds["Mesh2_face_nodes"].values
 
         res_face_nodes_connectivity = self._revert_edges_conn_to_face_nodes_conn(
             edge_nodes_connectivity, face_edges_connectivity,
             face_nodes_connectivity)
 
-        # Compare the res_face_nodes_connectivity with the uds._ds["Mesh2_face_nodes"].values
+        # Compare the res_face_nodes_connectivity with the uds.ds["Mesh2_face_nodes"].values
         self.assertTrue(
             np.array_equal(res_face_nodes_connectivity,
-                           uds._ds["Mesh2_face_nodes"].values))
+                           uds.ds["Mesh2_face_nodes"].values))
+
+
+class TestMeshLoader(TestCase):
+
+    def test_from_ds(self):
+        exodus_filepath = current_path / "meshfiles" / "exodus" / "outCSne8" / "outCSne8.g"
+
+        xrds_exodus = xr.open_dataset(exodus_filepath)
+
+        # Create the grid with multi-precision off
+        gridOff = ux.Grid(xrds_exodus)
```

Comment:
The `ux.Grid(xrds_exodus)` should call the `Grid.__from_ds__()` internally.

Also, please put this initializing from ds testing inside the previous `class testGrid` and when you are testing for the  multi-precision, name your function similar to `def test_from_ds_multiprecision`

Reply:
I am not sure what you mean when you say it should call it internally. Isn't that what I am doing in line 888?

Reply:
Since `ux.Grid(xrds_exodus)` should  already call the `Grid.__from_ds__()` internally. The line 888 is a duplicate call